tests/unit_tests/test_engine.py

```
@@ -367,3 +387,1790 @@ def test_add_health_check_not_callable(test_app: Engine):
+# Notifier Capability Cache Tests
+# =============================================================================
+
+
+class TestNotifierCapabilityCache:
+    """Tests for notifier capability caching."""
+
+    def test_cache_avoids_repeated_signature_inspection(self, test_app: Engine):
+        """Test that signature inspection is only called once per notifier."""
+        from unittest.mock import patch
+
+        def notifier(_message: str, attachments: list[Attachment] | None = None) -> None:  # noqa: ARG001
```

Ruff (0.14.13)

Unused `noqa` directive (unused: `ARG001`)

Remove unused `noqa` directive

(RUF100)